src/verification/others.rs

```diff
@@ -20,16 +20,35 @@ pub struct Verifier {
     /// Testing only: an additional root CA certificate to trust.
     #[cfg(any(test, feature = "ffi-testing", feature = "dbg"))]
     test_only_root_ca_override: Option<Vec<u8>>,
+
+    #[cfg(feature = "extra-roots")]
+    extra_roots: Vec<rustls::OwnedTrustAnchor>,
 }
 
 impl Verifier {
     /// Creates a new verifier whose certificate validation is provided by
-    /// WebPKI.
+    /// WebPKI, using root certificates provided by the platform.
     pub fn new() -> Self {
         Self {
             inner: OnceCell::new(),
             #[cfg(any(test, feature = "ffi-testing", feature = "dbg"))]
             test_only_root_ca_override: None,
+            #[cfg(feature = "extra-roots")]
+            extra_roots: Vec::new(),
+        }
+    }
+
+    /// Creates a new verifier whose certificate validation is provided by
+    /// WebPKI, using root certificates provided by the platform and augmented by
+    /// the provided extra root certificates.
+    #[cfg(feature = "extra-roots")]
+    pub fn new_with_extra_roots(roots: impl Iterator<Item = rustls::OwnedTrustAnchor>) -> Self {
```

Comment:
Would it be better to use `impl IntoIterator` here?

Reply:
Switched, thanks!